sup.go

```diff
@@ -92,6 +94,18 @@ func (sup *Stackup) Run(network *Network, commands ...*Command) error {
 		}
 	}
 
+	// if any cmd needs sudo preload the password so it's not needed 3 commands in
+	var pass string
+	for _, cmd := range commands {
+		if cmd.Sudo {
+			var err error
+			pass, err = prompt(fmt.Sprintf("sudo password for hosts:"))
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
We can't read password like this every single time "sudo" is defined. You can even have sudo w/o any password.